src/lib/common/sol-platform-linux-common.c

```diff
@@ -986,7 +986,11 @@ sol_platform_impl_load_locales(char **locale_cache)
 
         category = system_locale_to_sol_locale(key);
         if (category != SOL_PLATFORM_LOCALE_UNKNOWN) {
-            locale_cache[category] = sol_str_slice_to_string(value);
+            struct sol_buffer buf;
+
+            r = sol_util_escape_quotes(value, &buf);
+            SOL_INT_CHECK_GOTO(r, < 0, err_val);
+            locale_cache[category] = sol_buffer_steal_or_copy(&buf, NULL);
```

Comment:
although you are doing steal-or-copy, make sure you call `sol_buffer_fini()` to keep the pattern. I'm worried we're skipping fini because we know there is no memory left in it to be freed, but this is bad development pattern since people may read our code and use it as example and do mistake in theirs.

This is something we should also review in that batch from you where you moved some APIs to buffer. Just to make it consistent.